adb_shell/adb_device.py

```diff
@@ -425,6 +458,30 @@ def shell(self, command, timeout_s=None, total_timeout_s=constants.DEFAULT_TOTAL
         """
         return self._service(b'shell', command.encode('utf8'), timeout_s, total_timeout_s, decode)
 
+    def streaming_shell(self, command, timeout_s=None, total_timeout_s=constants.DEFAULT_TOTAL_TIMEOUT_S, decode=True):
```

Comment:
Please add tests for this method with `decode` set to both `True` and `False` so that `_streaming_service` and this function have full coverage.